cpp/include/cudf/detail/gather.cuh

```diff
@@ -83,8 +83,7 @@ struct gather_bitmask_functor {
     auto col     = input.column(mask_idx);
 
     if (Op != gather_bitmask_op::DONT_CHECK) {
-      bool out_of_range = is_signed_iterator<MapIterator>() ? (row_idx < 0 || row_idx >= col.size())
-                                                            : row_idx >= col.size();
+      bool out_of_range = (row_idx < 0 || row_idx >= col.size());
```

Comment:
Pretty confident that this is going to cause a warning about comparison with zero for unsigned types. That's likely why the `is_signed` is there.

Reply:
Actually, the indexalator normalizes all indices to `size_type` so this check is no longer needed.

Reply:
That assumes all iterators passed to this function are via the indexalator. I don't think we can make that assumption.

Reply:
That is a good point. Someone could pass in unsigned indices since this is in a detail namespace.
I will put it back in. I think the `is_signed_iterator` should be defined in `traits.hpp`.